realm/realm-library/src/main/java/io/realm/RealmConfiguration.java

```diff
@@ -132,6 +139,39 @@ protected RealmConfiguration(@Nullable File realmDirectory,
         this.initialDataTransaction = initialDataTransaction;
         this.readOnly = readOnly;
         this.compactOnLaunch = compactOnLaunch;
+        this.forceSyncHistory = forceSyncHistory;
+    }
+
+    /**
+     * Returns a SyncConfiguration appropriate to open a synced Realm offline.
+     * This is useful when trying to open a backup Realm (after a Client Reset)
+     *
+     * @param canonicalPath the absolute path to the Realm file defined by this configuration.
+     * @param modules if specified it will restricts Realm schema to the provided module.
+     * @return SyncConfiguration that can be used offline
+     */
+    public static RealmConfiguration forOffline(String canonicalPath, Object... modules) {
```

Comment:
wouldn't it be enough to have this in `SyncConfiguration`?
Another thing is this doesn't cover all cases.
eg.: The sync Realm is encrypted, the backup realm file is still encrypted. So we still need to give user a chance to add more params to the config.
Would it make sense to return a `Builder` instead?

Reply:
I think it's consistent to build the offline conf from `RealmConfiguration` otherwise it will be confusing to have an offline SyncConf.

> The sync Realm is encrypted, the backup realm file is still encrypted.
Will add a test to cover this use case

Reply:
cocoa has this api in sync config extension https://github.com/realm/realm-cocoa/pull/5259/files#diff-89bce12b01636fffc4e7d1848568b610R72

`RealmConfiguration.forOffLine()` sounds strange to me, but i don't have a better name + i am idiot of naming something ... so if @realm/java is fine with this, i am fine.

Reply:
It would be odd to see Sync things if Sync is disabled.

Reply:
@beeender Unfortunately we can't use the builder as it is now since we don't have the option like in Cocoa for instance where you can pass directly the location of the Realm (`fileURL`) we can combine `directory` and `name` option on the builder, whereas the ObjectStore return the full path to the backup file

I will revert the option `forOffline` on `SyncConfiguration`.